### PR TITLE
Test Local functions against date command

### PR DESCRIPTION
**Pull request by esheppa**
This is split from #678 and just adds the extra tests

### CODE REVIEW

File: src/offset/local/mod.rs [hunk 1]
```diff
@@ -137,7 +137,149 @@ impl TimeZone for Local {
...
+    }
+
+    #[test]
+    #[cfg(unix)]
+    fn try_verify_against_date_command() {
+        // #TODO: investigate /bin/date command behaviour on macOS
+        // avoid running this on macOS, temporarily
+        // for date_path in ["/usr/bin/date", "/bin/date"] {
+        for date_path in ["/usr/bin/date"] {
+            if path::Path::new(date_path).exists() {
```
Comment (djc) [inline src/offset/local/mod.rs:200]: Can you make this use an early continue? That would help reduce rightward drift.

**Review by djc — COMMENTED** [review]
Looks great!

File: src/offset/local/mod.rs (hunk 1 above)
Comment (esheppa) [reply src/offset/local/mod.rs]: good call, the code was a bit gnarly, I've made some improvements by using NaiveDateTime instead

Comment (esheppa) [reply src/offset/local/mod.rs]: I removed the loop across date commands as I tried this out on macOS, but I couldn't find the same command line arguments and so wasn't able the replicate the behavior desired to find the local timezone

**Pull request merged**